src/main/java/com/mongodb/quickstart/MappingPOJO.java

```diff
@@ -0,0 +1,65 @@
+package com.mongodb.quickstart;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.FindOneAndReplaceOptions;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.quickstart.models.Grade;
+import com.mongodb.quickstart.models.Score;
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.mongodb.client.model.Filters.eq;
+import static java.util.Collections.singletonList;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+public class MappingPOJO {
+
+    public static void main(String[] args) {
+        Logger.getLogger("org.mongodb.driver").setLevel(Level.WARNING);
+        ConnectionString connectionString = new ConnectionString(System.getProperty("mongodb.uri"));
+        CodecRegistry pojoCodecRegistry = fromProviders(PojoCodecProvider.builder().automatic(true).build());
+        CodecRegistry codecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(), pojoCodecRegistry);
+        MongoClientSettings clientSettings = MongoClientSettings.builder()
+                                                                .applyConnectionString(connectionString)
+                                                                .codecRegistry(codecRegistry)
+                                                                .build();
+        try (MongoClient mongoClient = MongoClients.create(clientSettings)) {
+            MongoDatabase db = mongoClient.getDatabase("sample_training");
+            MongoCollection<Grade> grades = db.getCollection("grades", Grade.class);
+
+            // create a new grade.
+            Grade newGrade = new Grade().setStudent_id(10003d)
+                                        .setClass_id(10d)
+                                        .setScores(singletonList(new Score().setType("homework").setScore(50d)));
+            grades.insertOne(newGrade);
+
+            // find this grade.
+            Grade grade = grades.find(eq("student_id", 10003d)).first();
+            System.out.println("Grade found:\t" + grade);
+
+            // update this grade: adding an exam grade
```

Comment:
Sure we don't just want to do an update and push a new score onto the array?

Reply:
If I do that then I have to manipulate "Document" again. I covered this in the update blog post. Here I'm trying to show how to make it using only POJOs. 
But yes, it's less optimal using POJOs, I totally agree but that's the trade off for using POJOs in this case of situation.